src/main/java/com/aws/greengrass/deployment/DefaultDeploymentTask.java

```diff
@@ -111,6 +115,27 @@ public DeploymentResult call() throws InterruptedException {
                     .kv("Deployment service config", deploymentServiceConfig.toPOJO().toString())
                     .log("Starting deployment task");
 
+            // Enforce deployment sequence validation using timestamps:
+            // If we have already processed a more recent deployment for a thing group, then we will drop
+            // this deployment and only process it if it is newer (or same) deployment for that thing group.
+            if (isStaleDeployment(deploymentDocument)) {
+                logger.atInfo().setEventType(DEPLOYMENT_TASK_EVENT_TYPE)
+                        .log("Nucleus has a newer deployment for '{}' target. Rejecting the deployment",
+                                deploymentDocument.getGroupName());
+
+                Topics lastDeployment =
+                        deploymentServiceConfig.lookupTopics(DeploymentService.GROUP_TO_LAST_DEPLOYMENT_TOPICS)
+                                .lookupTopics(deploymentDocument.getGroupName());
```

Comment:
you don't need 2 calls to lookup topics, just use `lookupTopics(GROUP, getGroupName())`

Reply:
I didn't know that :)